### DISCUSSION TITLE
Android setup problem

### DISCUSSION THREAD

**Discussion by HaronK** [slint-ui/slint#5662] [Q&A] [answered]
Hi!
I was looking for gui library that could build Android applications and looks like Slint is most advanced here.
So I followed tutorial and got a nice memory game. After that I tried to configure it to build for Android using instructions from [here](https://snapshots.slint.dev/master/docs/rust/slint/android/).
Everything was compiling up until **i-slint-backend-android-activity v1.7.0**:
```
error: failed to run custom build command for `i-slint-backend-android-activity v1.7.0`

Caused by:
  process didn't exit successfully: `/mnt/WinF/Personal/workspace/rust/memory/target/debug/build/i-slint-backend-android-activity-2829fd63d41cb1f9/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=ANDROID_HOME
  cargo:rerun-if-env-changed=JAVA_HOME

  --- stderr
  thread 'main' panicked at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/i-slint-backend-android-activity-1.7.0/build.rs:83:9:
  Java compilation failed: javac: directory not found: /mnt/WinF/Personal/workspace/rust/memory/target/aarch64-linux-android/debug/build/i-slint-backend-android-activity-ad21d1c17511365e/out/java/dev/slint/android-activity
  Usage: javac <options> <source files>
  use -help for a list of possible options

  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
Error: Command `AR_aarch64-linux-android=/home/user/Android/Sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-ar CARGO_ENCODED_RUSTFLAGS=-Clink-arg=--target=aarch64-linux-android23\u{1f}-L\u{1f}/mnt/WinF/Personal/workspace/rust/memory/target/cargo-apk-temp-extra-link-libraries CARGO_TARGET_AARCH64_LINUX_ANDROID_AR=/home/user/Android/Sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/bin/llvm-ar CARGO_TARGET_AARCH64_LINUX_ANDROID_LINKER=/home/user/Android/Sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/bin/clang CC_aarch64-linux-android=/home/user/Android/Sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/bin/clang CFLAGS_aarch64-linux-android=--target=aarch64-linux-android23 CXXFLAGS_aarch64-linux-android=--target=aarch64-linux-android23 CXX_aarch64-linux-android=/home/user/Android/Sdk/ndk/27.0.12077973/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ cargo build --lib --target aarch64-linux-android` had a non-zero exit code.
```
I tried to compile another Slint Android project mentioned in discussions here - [rssbox-android](https://github.com/Heng30/rssbox-android) and got similar error, so I assume it's some Android dev environment configuration problem.

Do I miss something? Any ideas?
Thanks in advance.

**Comment by ogoffart** [comment]
Hi,
Not sure what is causing the problem.
What is your version of javac as reported by `javac -version`?
Do the directory /mnt/WinF/Personal/workspace/rust/memory/target/aarch64-linux-android/debug/build/i-slint-backend-android-activity-ad21d1c17511365e/out/java/dev/slint/android-activity not exist?
(maybe i need to create the output directory in advance, but i didn't think it was necessary)

**Reply by HaronK** [reply]
Hi!
1. javac 1.8.0_412
2. exists only **/mnt/WinF/Personal/workspace/rust/memory/target/aarch64-linux-android/debug/build/i-slint-backend-android-activity-ad21d1c17511365e/out/** folder that is empty

**Reply by ogoffart** [reply, marked as answer]
Java 1.8 is a bit old, but it should in theory work. I've added code in https://github.com/slint-ui/slint/pull/5672 to create this directory before so that should solve that problem in the next version.
In the mean time, you can try upgrade java to JDK 17,  (but not 21 as it has other problem)

**Reply by HaronK** [reply]
Great! With your patch and updating to JDK 17, apk was built. Thanks!
